internal/topology/variables/cluster_variable_defaulting.go

```diff
@@ -83,9 +83,6 @@ func defaultClusterVariable(clusterVariable *clusterv1.ClusterVariable, clusterC
 
 	var value interface{}
 	// If the schema has a default value, default it.
-	// NOTE: This is essentially the same which the structural schema defaulting
-	// lib would do, except that it wouldn't default nullable values.
-	// As we want to also default nullable variables we have to do it ourselves.
 	if apiExtensionsSchema.Default != nil {
```

Comment:
Let's keep the simple defaulting for this PR. The new full implementation is part of: https://github.com/kubernetes-sigs/cluster-api/pull/5779 (including object/array support)